assert/assert_array_includes.ts

```diff
@@ -19,8 +22,8 @@ import { AssertionError } from "./assertion_error.ts";
  * ```
  */
 export function assertArrayIncludes<T>(
-  actual: ArrayLike<T>,
-  expected: ArrayLike<T>,
+  actual: ArrayLikeArg<T>,
+  expected: ArrayLikeArg<T>,
   msg?: string,
 ): void {
```

Comment:
Infered `T` seems preventing users comparing arrays of incompatible types. I think this `T` is still useful.

<img width="897" alt="Screenshot 2023-12-08 at 17 38 24" src="https://github.com/denoland/deno_std/assets/613956/7476519c-efb8-4672-9a37-b20c200e35d8">

Reply:
@kt3k I agree that it could be helpful in some cases, but it can also produce misleading diagnostics which don't reflect runtime results. Here's one contrived example — I'm sure we could find others in real program code:

`test.ts`
```ts
import { assertArrayIncludes } from "https://deno.land/std@0.208.0/assert/assert_array_includes.ts";

Deno.test("deno_std pr 3910", () => {
  const foo = { foo: "foo" };
  const bar = { bar: "bar" };

  Object.defineProperty(foo, "bar", { value: "bar" });
  Object.defineProperty(bar, "foo", { value: "foo" });

  assertArrayIncludes([foo], [bar]);
});

```

Terminal:

```txt
% deno --version
deno 1.38.5 (release, aarch64-apple-darwin)
v8 12.0.267.1
typescript 5.2.2
```

Type-checking fails
```txt
% deno test test.ts
Check file:///Users/deno/test.ts
error: TS2741 [ERROR]: Property 'foo' is missing in type '{ bar: string; }' but required in type '{ foo: string; }'.
  assertArrayIncludes([foo], [bar]);
                              ~~~
    at file:///Users/deno/test.ts:10:31

    'foo' is declared here.
      const foo = { foo: "foo" };
                    ~~~~~~~~~~
        at file:///Users/deno/test.ts:4:17
```

…but the test actually passes
```txt
% deno test --no-check test.ts
running 1 test from ./test.ts
deno_std pr 3910 ... ok (0ms)

ok | 1 passed | 0 failed (0ms)

```

Reply:
@kt3k I see that perspective. I guess I'd still like to see actual statistical data to get a realistic idea of usage patterns. Perhaps I'll start by taking a look at GitHub's code search API.